modules/parsers/parser-hdbdd/src/main/java/com/sap/xsk/parser/hdbdd/symbols/context/ContextSymbol.java

```diff
@@ -61,4 +70,36 @@ public Map<String, Symbol> getSymbols() {
   public boolean isDuplicateName(String id) {
     return symbols.containsKey(id) || getName().equals(id);
   }
+
+  private Symbol findSymbol(String name) {
+    Scope currentScope = this;
+    Symbol resolvedSymbol = this.getSymbols().get(name);
+    while (resolvedSymbol == null) {
+      if (currentScope instanceof CDSFileScope) {
+        return null;
+      }
+
+      currentScope = currentScope.getEnclosingScope();
+      resolvedSymbol = currentScope.resolve(name);
+    }
+
+    return resolvedSymbol;
+  }
+
+  private Symbol resolveByFullSymbolName(String[] contexts, Symbol outerContext) {
+    if (outerContext instanceof ContextSymbol) {
```

Comment:
Could you reverse this condition to `if (!(outerContext instanceof ContextSymbol))` and return `null` there in order to reduce code nesting?